src/Game/UI/Gumps/WorldMapGump.cs

```diff
@@ -387,6 +395,57 @@ public void GoToMarker(int x, int y, bool isManualType)
             _center.Y = y;
         }
 
+        private void BuildContextMenuForZones(ContextMenuControl parent)
+        {
+            ContextMenuItemEntry zoneOptions = new ContextMenuItemEntry(ResGumps.MapZoneOptions);
+
+            // XXX Is it a good idea to call BuildContextMenu() from within an action stored in the context menu?
+            zoneOptions.Add(new ContextMenuItemEntry(ResGumps.MapZoneReload, () => { LoadZones(); BuildContextMenu(); }));
```

Comment:
Answering to your comment: I don't see any particular issue calling BuildContextMenu() into a callback. Give a try when you can